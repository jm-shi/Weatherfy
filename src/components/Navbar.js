import React from 'react';
import Searchbar from './Searchbar';
import { NavLink } from 'react-router-dom';

class Navbar extends React.Component {
  render() {
    const { onHomePage } = this.props;
    return (
      <div className="header">
        {onHomePage ? (
          <NavLink className="home logo" exact to="/">
            WEATHER
          </NavLink>
        ) : (
          <NavLink className="logo" exact to="/">
            WEATHER
          </NavLink>
        )}
        {onHomePage ? (
          <span />
        ) : (
          <div style={{ marginTop: '20px' }}>
            <Searchbar />
          </div>
        )}
      </div>
    );
  }
}

module.exports = Navbar;
