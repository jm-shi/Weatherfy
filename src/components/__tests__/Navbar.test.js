import React from 'react';
import Navbar from '../Navbar.js';
import { shallow } from 'enzyme';

test('should render Navbar correctly', () => {
  const wrapper = shallow(<Navbar />);
  expect(wrapper).toMatchSnapshot();
});
