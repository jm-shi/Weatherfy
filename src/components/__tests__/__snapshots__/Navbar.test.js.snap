// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Navbar correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="logo"
          exact={true}
          to="/"
        >
          WEATHER
        </NavLink>,
        <div
          style={
            Object {
              "marginTop": "20px",
            }
          }
        >
          <Searchbar />
        </div>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "WEATHER",
          "className": "logo",
          "exact": true,
          "to": "/",
        },
        "ref": null,
        "rendered": "WEATHER",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Searchbar />,
          "style": Object {
            "marginTop": "20px",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="logo"
            exact={true}
            to="/"
          >
            WEATHER
          </NavLink>,
          <div
            style={
              Object {
                "marginTop": "20px",
              }
            }
          >
            <Searchbar />
          </div>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "WEATHER",
            "className": "logo",
            "exact": true,
            "to": "/",
          },
          "ref": null,
          "rendered": "WEATHER",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Searchbar />,
            "style": Object {
              "marginTop": "20px",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
